src/components/options/SpcOptionEditor.tsx

```diff
@@ -28,24 +28,32 @@ export function SpcOptionEditor(props: Props) {
   const styles = useStyles2(getStyles);
 
   const { value: options, isVar } = useParseSpcOptions(props.value);
+  const [customOptions, setCustomOptions] = React.useState<Array<SelectableValue<number>>>([]);
   const onChange = props.onChange;
 
   return (
     <>
       <div className={styles.row}>
         <InlineField label="Sample size" disabled={isVar}>
           <Select
-            options={sampleSizeOptions}
+            options={[...sampleSizeOptions, ...customOptions]}
```

Comment:
Consider merging these two arrays with `useMemo` to prevent them from being regenerated on every re-render.

Reply:
My suggestion is:

```typescript
const [customOption, setCustomOption] = React.useState<SelectableValue<number> | null>(null);
// ..

const allOptions = React.useMemo(() => {
    if (customOption) {
      return [...sampleSizeOptions, customOption];
    }
    return sampleSizeOptions;
  }, [customOption]);
```